metadata/src/audio/item.rs

```diff
@@ -16,98 +18,290 @@ use librespot_core::{
+                            None
+                        }
+                    })
+                    .collect();
+
+                let alternatives = if track.alternatives.is_empty() {
+                    None
                 } else {
-                    return Ok(());
+                    Some(track.alternatives)
+                };
+
+                let availability = if Date::now_utc() < track.earliest_live_timestamp {
+                    Err(UnavailabilityReason::Embargo)
+                } else {
+                    available_for_user(
+                        &session.user_data(),
+                        &track.availability,
+                        &track.restrictions,
+                    )
+                };
+
+                let popularity = track.popularity.max(0).min(100) as u8;
+                let number = track.number.max(0) as u32;
+                let disc_number = track.disc_number.max(0) as u32;
+
+                let unique_fields = UniqueFields::Track {
+                    artists,
+                    album,
+                    album_artists,
+                    popularity,
+                    number,
+                    disc_number,
+                };
+
+                Ok(Self {
+                    track_id,
+                    uri,
+                    files: track.files,
+                    name: track.name,
+                    covers,
+                    language: track.language_of_performance.clone(),
+                    duration_ms: track.duration as u32,
+                    is_explicit: track.is_explicit,
+                    availability,
+                    alternatives,
+                    unique_fields,
+                })
+            }
+            SpotifyItemType::Episode => {
+                let episode = Episode::get(session, &id).await?;
+
+                if episode.duration <= 0 {
+                    return Err(Error::unavailable(MetadataError::InvalidDuration(
+                        episode.duration,
+                    )));
                 }
+
+                if episode.is_explicit {
+                    if let Some(value) = session.get_user_attribute("filter-explicit-content") {
+                        if &value == "1" {
+                            return Err(Error::unavailable(MetadataError::ExplicitContentFiltered));
+                        }
+                    }
+                }
+
+                let track_id = episode.id;
+                let uri = track_id.to_uri()?;
+
+                let mut covers = episode.covers;
+
+                covers.sort_by(|a, b| b.width.cmp(&a.width));
+
+                let covers = covers
+                    .iter()
+                    .filter_map(|cover| {
+                        let cover_id = cover.id.to_string();
+
+                        if !cover_id.is_empty() {
+                            let cover_image = CoverImage {
+                                url: image_url.clone().replace("{file_id}", &cover_id),
```

Comment:
Should have been suggested in the other review already; maybe GitHub is doing something funky. Here you go.

```suggestion
                                url: image_url.replace("{file_id}", &cover_id),
```

Reply:
Thanks.